src/OctopusPuppet.Tests/DeploymentPlanActionTests.cs

```diff
@@ -0,0 +1,129 @@
+                new[]
+                {
+                    new ReleaseResource(versionNumber, "124", "")
+                    {
+                        Id = versionNumber,
+                        Version = versionNumber,
+                        ProjectVariableSetSnapshotId = "fake-variable-set-id",
+                        ProjectDeploymentProcessSnapshotId = Guid.NewGuid().ToString()
+                    }
+                },
+                new LinkCollection()
+            ));
+
+            //  Environments 
+            var envRepo = Substitute.For<IEnvironmentRepository>();
+            envRepo.GetAll().Returns(_ => new List<ReferenceDataItem>
+            {
+                new ReferenceDataItem("124", environmentName)
+            });
+            repo.Environments.Returns(envRepo);
+
+            // Dashboard 
+            var dashboardRepo = Substitute.For<IDashboardRepository>();
+            dashboardRepo.GetDynamicDashboard(Arg.Any<string[]>(), Arg.Any<string[]>())
+                .Returns(_ => new DashboardResource
+                {
+                    Projects = new List<DashboardProjectResource>
+                    {
+                        new DashboardProjectResource { Id = "124", Name = "arminfra" }
+                    },
+                    Items = new List<DashboardItemResource>
+                    {
+                        new DashboardItemResource
+                        {
+                            ProjectId = "124",
+                            EnvironmentId = environmentName,
+                            ReleaseVersion = versionNumber,
+                            State = TaskState.Success,
+                            QueueTime = DateTime.UtcNow.AddMinutes(-5),
+                            CompletedTime = DateTime.UtcNow
+                        }
+                    }
+                });
+            repo.Dashboards.Returns(dashboardRepo);
+
+            // Variable Sets
+            var variableSetRepo = Substitute.For<IVariableSetRepository>();
+            variableSetRepo.Get(Arg.Any<string>()).Returns(_ => new VariableSetResource
+            {
+                Variables = new List<VariableResource>
+                {
+                    new VariableResource { Id = Guid.NewGuid().ToString(), Name = "test", Value = "Testing var" }
+                }
+            });
+            repo.VariableSets.Returns(variableSetRepo);
+
+            return new OctopusDeploymentPlanner(repo);
+        }
+        [Fact]
+        public void DeploymentPlanner_Should_SkipMaster_WhenFlagIsTrue()
+        {
+            var planner = GetSutForVersionTest("1.2.3456", "D1");
+
+            var result = planner.GetBranchDeploymentPlans("D1", "1.2.3456", false, true,
+                new ComponentFilter
+                {
+                    Expressions = new List<string> { "(?i)^(ArmSharedInfrastructure|Filebeat)$" }
```

Comment:
This doesn't seem to be exercised -probably something missing from mock. Agreed to create a new test to verify expression filter works (e.g. look for 2 out of 3 projects) to find out what's missing from mock.